x-pack/plugins/transform/public/app/sections/transform_management/components/transform_list/use_table_settings.ts

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { useState } from 'react';
+import { Direction, EuiBasicTableProps, EuiTableSortingType } from '@elastic/eui';
+import sortBy from 'lodash/sortBy';
+import get from 'lodash/get';
+import { TransformListRow } from '../../../../common';
+
+const PAGE_SIZE = 10;
+const PAGE_SIZE_OPTIONS = [10, 25, 50];
+
+const propertyMap = {
+  Mode: 'mode',
+};
+
+interface AnalyticsBasicTableSettings {
+  pageIndex: number;
+  pageSize: number;
+  totalItemCount: number;
+  hidePerPageOptions: boolean;
+  sortField: string;
+  sortDirection: Direction;
+}
+
+interface UseTableSettingsReturnValue {
+  onTableChange: EuiBasicTableProps<TransformListRow>['onChange'];
+  pageOfItems: TransformListRow[];
+  pagination: EuiBasicTableProps<TransformListRow>['pagination'];
+  sorting: EuiTableSortingType<any>;
+}
+
+export function useTableSettings(
+  sortByField: string,
+  items: TransformListRow[]
+): UseTableSettingsReturnValue {
+  const [tableSettings, setTableSettings] = useState<AnalyticsBasicTableSettings>({
+    pageIndex: 0,
+    pageSize: PAGE_SIZE,
+    totalItemCount: 0,
+    hidePerPageOptions: false,
+    sortField: sortByField,
+    sortDirection: 'asc',
+  });
+
+  const getPageOfItems = (
+    list: any[],
```

Comment:
Can this be `TransformListRow[]`?

Reply:
Updated `useTableSettings` type to be generic and accept item type - 9c2d7d1935a8d069b9df63af19dade005a99e624